### PR TITLE
fix: class-scope field annotation typeMap keys to prevent cross-class collision

### LINKED ISSUE #1458: bug: class-field annotation typeMap keys are not class-scoped (cross-class collision)
## Problem

Both extractors (TS \`handleFieldDefTypeMap\` in \`src/extractors/javascript.ts\`, Rust \`public_field_definition | field_definition\` branch in \`crates/codegraph-core/src/extractors/javascript.rs\`) seed class-field type annotations as **bare file-wide keys**:

```ts
class OrderService {
  private repo: OrderRepository;   // seeds: repo → OrderRepository, this.repo → OrderRepository
}
class UserService {
  private repo: UserRepository;    // seeds: repo → UserRepository, this.repo → UserRepository (collides)
}
```

Both entries land at confidence 0.9, so first-wins-on-tie semantics keep \`OrderRepository\` for *both* classes — \`this.repo.save()\` inside \`UserService\` resolves to \`OrderRepository.save\`.

This is the same collision class as issue #1323 (constructor property writes), which was fixed by class-scoping the key as \`ClassName.prop\`. Field annotations should use the same scheme: seed \`ClassName.field\` (resolver already tries the \`CallerClass.X\` fallback added in #1382), keeping the bare key only as a lower-priority fallback or dropping it.

## Scope note

Both engines currently implement the identical (colliding) behavior, so this is parity-neutral — filed as a correctness follow-up found while implementing the engine-parity fixes for #1453.

## Fix sketch

1. In both extractors, compute the enclosing class name (TS: \`childTypeMapClass\` context already threaded through the walk; Rust: \`enclosing_type_map_class\` helper already exists).
2. Seed \`ClassName.field\` instead of (or in addition to) bare \`field\` / \`this.field\`.
3. Update resolver expectations/tests in both engines together to preserve parity.

### PR DESCRIPTION
**Pull request by carlos-alm**
Field type annotations (`private repo: OrderRepository`) were seeded as bare file-wide typeMap keys, causing `this.repo` inside `UserService` to resolve to `OrderRepository` when both classes had a `repo` field.

Both extractors (TS `handleFieldDefTypeMap` and Rust `field_definition` branch) now seed `ClassName.field` keys at confidence 0.9, matching the `CallerClass.X` resolver fallback added in PR #1382. Bare keys are kept at confidence 0.6 as fallbacks for single-class files or class expressions where no enclosing class name is available.

Both engines change identically — parity preserved.

Closes #1458

### CODE REVIEW

File: tests/parsers/javascript.test.ts [hunk 1]
```diff
@@ -217,15 +217,34 @@ describe('JavaScript parser', () => {
...
+        class UserService {
+          private repo: UserRepository;
+        }
+      `);
+      // Each class gets its own scoped key — no collision.
+      expect(symbols.typeMap.get('OrderService.repo')).toEqual({ type: 'OrderRepository', confidence: 0.9 });
+      expect(symbols.typeMap.get('UserService.repo')).toEqual({ type: 'UserRepository', confidence: 0.9 });
+      // Bare "repo" key should hold the first class's type at 0.6 (second write is same confidence, no overwrite).
+      expect(symbols.typeMap.get('repo')?.confidence).toBe(0.6);
     });
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply tests/parsers/javascript.test.ts:257]: Fixed. The resolver (call-resolver.ts) now checks the class-scoped key (`ClassName.prop`) before bare fallback keys for `this.` receivers. Added integration test `issue-1458-cross-class-field-typemap.test.ts` that calls `buildGraph` with both classes in the same file and asserts OrderService.run → OrderRepository.save and UserService.run → UserRepository.save, with no false cross-class edges.

File: crates/codegraph-core/src/extractors/javascript.rs [hunk 2]
```diff
@@ -216,9 +221,33 @@ fn match_js_type_map(node: &Node, source: &[u8], symbols: &mut FileSymbols, _dep
...
+                                }
+                                None => {
+                                    // No enclosing class declaration (e.g. class expression)
+                                    // — use bare keys only at full confidence.
+                                    push_type_map_entry(symbols, field_name.clone(), type_name.to_string());
+                                    push_type_map_entry(symbols, format!("this.{}", field_name), type_name.to_string());
+                                }
+                            }
                         }
                     }
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply crates/codegraph-core/src/extractors/javascript.rs:252]: Added `field_annotation_multi_class_seeds_separate_scoped_keys` Rust unit test in `crates/codegraph-core/src/extractors/javascript.rs`. It parses two classes with identically-named fields and asserts `OrderService.repo → OrderRepository @ 0.9` and `UserService.repo → UserRepository @ 0.9` are both present as separate entries — mirroring the TS collision test.

**Comment by carlos-alm** [comment]
Addressed Greptile's findings:

**Root cause confirmed:** The resolver checked bare `repo` (0.6) before class-scoped `ClassName.repo` (0.9), so the class-scoped fix in the extractor had no effect at resolution time.

**Fixes pushed:**
1. `fix(resolver): check class-scoped typeMap key before bare fallback for this. receivers` — both `call-resolver.ts` and `build_edges.rs` now probe the class-scoped key first for `this.` receivers when `callerName` is available.
2. `test(1458): add Rust multi-class field collision unit test and end-to-end integration test` — Rust unit test mirrors the TS collision test; integration test drives the full WASM build path and asserts correct/absent edges for both classes.

**Lint:** was failing due to Biome line-length formatting on the new test assertions — already fixed in the original commit; confirmed passing now.

All 4 new integration test assertions pass. Rust compiles clean (`cargo check`).

**Comment by carlos-alm** [comment]
@greptileai

File: crates/codegraph-core/src/extractors/javascript.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlos-alm) [reply crates/codegraph-core/src/extractors/javascript.rs:252]: The test was using parse_js() (tree-sitter-javascript) but the fixture uses TypeScript-specific syntax (private field declarations with type annotations). The JS grammar doesn't support type annotations so the type_map came back empty, causing the assertion to fail on CI. Fixed by adding parse_ts() using LANGUAGE_TYPESCRIPT and switching the test to it. All 396 Rust unit tests now pass locally.

**Comment by carlos-alm** [comment]
Fix for CI failures: The `field_annotation_multi_class_seeds_separate_scoped_keys` Rust test was using `parse_js()` (tree-sitter-javascript) but the fixture uses TypeScript-specific syntax (`private repo: TypeAnnotation`). The JS grammar does not support type annotations, so the `type_map` came back empty and the test failed on all 3 platforms.

Fixed by adding a `parse_ts()` helper (using `LANGUAGE_TYPESCRIPT`) to the test module and switching the test to it. All 396 Rust unit tests pass locally.

Out-of-scope findings from the Claude review filed as GitHub issues:
- #1500: Missing test coverage for class expression (None path) in `enclosing_type_map_class`
- #1501: `push_type_map_entry` lacks first-write-wins parity with TS `setTypeMapEntry`

**Comment by carlos-alm** [comment]
CI fix (commit 76a4380): the Rust multi-class collision test was using `parse_js()` but the fixture uses TypeScript syntax (private field declarations with type annotations). Switched to `parse_ts()` — all 396 Rust unit tests now pass. No logic change.

**Comment by carlos-alm** [comment]
@greptileai
